src/compute-client/src/plan/interpret/api.rs

```diff
@@ -46,7 +46,7 @@ use crate::plan::{AvailableCollections, GetPlan, Plan};
 ///     <https://www.cs.utexas.edu/~wcook/Drafts/2012/ecoop2012.pdf>
 /// [tagless final encoding]: <https://okmij.org/ftp/tagless-final/>
 pub trait Interpreter<T = mz_repr::Timestamp> {
-    type Domain: Debug + Sized;
+    type Domain: Debug + Default + Sized;
```

Comment:
Why is the `Domain` bound required here?

Reply:
It's possible to get rid of it there if we just implement `Default` explicitly for `InterpreterContext`. I did that instead, so that we maintain less requirements for `Domain`.